Neos.FluidAdaptor/Classes/Core/ViewHelper/TemplateVariableContainer.php

```diff
@@ -13,32 +14,77 @@
 
 use Neos\FluidAdaptor\Core\Parser\SyntaxTree\TemplateObjectAccessInterface;
 use TYPO3Fluid\Fluid\Core\Variables\StandardVariableProvider;
-use TYPO3Fluid\Fluid\Core\Variables\VariableProviderInterface;
 
 /**
  * Provides the variables inside fluid template. Adds TemplateObjectAccessInterface functionality.
  *
  * @api
  */
-class TemplateVariableContainer extends StandardVariableProvider implements VariableProviderInterface
+class TemplateVariableContainer extends StandardVariableProvider
 {
     /**
      * Get a variable by dotted path expression, retrieving the
      * variable from nested arrays/objects one segment at a time.
      *
+     * This sadly mostly copies the parent method to add handling for
+     * subjects of type TemplateObjectAccessInterface.
+     *
      * @param string $path
      * @return mixed
      */
     public function getByPath($path)
     {
-        $subject = parent::getByPath($path);
+        // begin copy of parent method
+        $subject = $this->variables;
+        $subVariableReferences = explode('.', $this->resolveSubVariableReferences($path));
+        foreach ($subVariableReferences as $pathSegment) {
+            // begin TemplateObjectAccessInterface handling
+            if ($subject instanceof TemplateObjectAccessInterface) {
+                $subject = $subject->objectAccess();
+            }
+            // end TemplateObjectAccessInterface handling
+            if ((is_array($subject) && array_key_exists($pathSegment, $subject))
+                || ($subject instanceof \ArrayAccess && $subject->offsetExists($pathSegment))
+            ) {
+                $subject = $subject[$pathSegment];
+                continue;
+            }
+            if (is_object($subject)) {
+                $upperCasePropertyName = ucfirst($pathSegment);
+                $getMethod = 'get' . $upperCasePropertyName;
+                if (method_exists($subject, $getMethod)) {
+                    $subject = $subject->$getMethod();
+                    continue;
+                }
+                $isMethod = 'is' . $upperCasePropertyName;
+                if (method_exists($subject, $isMethod)) {
+                    $subject = $subject->$isMethod();
+                    continue;
+                }
+                $hasMethod = 'has' . $upperCasePropertyName;
```

Comment:
is it worth a consideration to use object access here?
And also can it be that we lost certain features over time in fluid like https://github.com/neos/flow-development-collection/pull/108?
I cant really judge and am not that invested into it and neither know whats best. But it seems its a bit back and forth ... we seemed to had already such manual implementation once and it got lost it seems?

Reply:
I did very consciously copy the method as unchanged as possible, to make the difference clearly visible and easy to understand. This will make maintenance easier for our future selves.

About #108 – yes, it seems that was lost. But given the decline of Fluid, I have little interest in fixing that nowish, and certainly not in the scope of this PR…